components/dashboard/src/teams/TeamUsage.tsx

```diff
@@ -82,6 +84,33 @@ function TeamUsage() {
         return totalCredits;
     };
 
+    const handleMonthClick = (start: any, end: any) => {
+        setStartDateOfBillMonth(start);
+        setEndDateOfBillMonth(end);
+    };
+
+    const getBillingHistory = () => {
+        let rows = [];
+        // This goes back 6 months from the current month
+        for (let i = 1; i < 7; i++) {
+            const endDateVar = i - 1;
+            const startDate = new Date(today.getFullYear(), today.getMonth() - i);
+            const endDate = new Date(today.getFullYear(), today.getMonth() - endDateVar, 0);
```

Comment:
Using this might make it easier logic vise. https://stackoverflow.com/questions/41505492/how-to-subtract-one-month-using-moment-js

I suspect the current logic has a bug for when 7 months back goes back to the previous year, as then you need to wrap the month and subtract 1 year

Reply:
You can test it by changing it to generate 12 months, or even 24 to see it

Reply:
> I suspect the current logic has a bug for when 7 months back goes back to the previous year, as then you need to wrap the month and subtract 1 year

It actually computes it as expected! I tried going back 12 months from July 2022 and it correctly shows me as July 2021.

Reply:
I'm surprised! But the JS version of Date, which is based on Java 1.0 package actually allows you to subtract from dates by using negative months!

```
var now = new Date()
now
Wed Jul 27 2022 11:02:40 GMT+0200 (Central European Summer Time)

new Date(now.getFullYear(), now.getMonth() - 10, 0)
Tue Aug 31 2021 00:00:00 GMT+0200 (Central European Summer Time)
```